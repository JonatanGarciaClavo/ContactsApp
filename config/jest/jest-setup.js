/*  eslint-disable */
// Make Enzyme functions available in all test files without importing
import { shallow, render, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
global.shallow = shallow;
global.render = render;
global.render = mount;
global.shallowToJson = shallowToJson;

// Skip createElement warnings but fail tests on any other warning
console.error = message => {
  if (!/(React.createElement: type should not be null)/.test(message)) {
      throw new Error(message);
  }
};