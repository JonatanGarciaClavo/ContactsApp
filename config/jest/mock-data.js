export const contact = {
  id: '-1',
  email: 'test@test.com',
  imgUrl: 'http://www.test.image.com',
  name: 'Tester',
  phoneNumber: '666666666',
};

export const contacts = [
  {
    id: '-1',
    email: 'test@test.com',
    imgUrl: 'http://www.test.image.com',
    name: 'Tester',
    phoneNumber: '666666666',
  },
  {
    id: '-2',
    email: 'test2@test.com',
    imgUrl: 'http://www.test2.image.com',
    name: 'Tester2',
    phoneNumber: '766666666',
  },
];

export const id = '-1';
