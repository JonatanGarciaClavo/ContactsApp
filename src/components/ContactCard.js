import React from 'react';
import { Card, CardMedia, CardTitle, CardActions, FlatButton } from 'material-ui';
// https://imgur.com/mbZIBzc
const ContactCard = ({ contact, onEditClick, onDeleteClick }) => {
  const { name, imgUrl, email, phoneNumber } = contact;
  return (
    <Card style={{ width: '300px', margin: '1em 0 0 0' }}>
      <CardMedia overlay={<CardTitle title={name} />}>
        <img src={imgUrl ||
          'http://i.imgur.com/mbZIBzc.png'}
        />
      </CardMedia>
      <CardTitle
        titleStyle={{ fontSize: '12pt' }}
        title={email}
        subtitle={phoneNumber || 'Without phone'}
      />
      <CardActions>
        <FlatButton onClick={onEditClick} onTouchStart={onEditClick} label="Edit" />
        <FlatButton onClick={onDeleteClick} onTouchStart={onDeleteClick} label="Delete" />
      </CardActions>
    </Card>
  );
}

ContactCard.propTypes = {
  contact: React.PropTypes.object.isRequired,
  onEditClick: React.PropTypes.func,
  onDeleteClick: React.PropTypes.func,
};

export default ContactCard;
