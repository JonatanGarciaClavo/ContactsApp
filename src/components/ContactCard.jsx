import React from 'react';
import { Card, CardMedia, CardTitle, CardActions, FlatButton } from 'material-ui';
import { Map } from 'immutable';
import _ from 'lodash';
// https://imgur.com/mbZIBzc
const ContactCard = ({ contact, onEditClick, onDeleteClick }) => {
  const { name, imgUrl, email, phoneNumber } = contact;
  return (
    <Card style={{ width: '300px', margin: '1em 0 0 0' }}>
      <CardMedia overlay={<CardTitle title={name} />}>
        <img
          alt=""
          role="presentation"
          src={imgUrl ||
            'http://i.imgur.com/mbZIBzc.png'}
        />
      </CardMedia>
      <CardTitle
        titleStyle={{ fontSize: '12pt' }}
        title={email}
        subtitle={phoneNumber || 'Without phone'}
      />
      <CardActions>
        <FlatButton onClick={onEditClick} onTouchStart={onEditClick} label="Edit" />
        <FlatButton onClick={onDeleteClick} onTouchStart={onDeleteClick} label="Delete" />
      </CardActions>
    </Card>
  );
}

ContactCard.defaultProps = {
  onEditClick: _.noop,
  onDeleteClick: _.noop,
}

ContactCard.propTypes = {
  contact: React.PropTypes.instanceOf(Map).isRequired,
  onEditClick: React.PropTypes.func,
  onDeleteClick: React.PropTypes.func,
};

export default ContactCard;
