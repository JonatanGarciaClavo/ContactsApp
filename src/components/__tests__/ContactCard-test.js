import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import _ from 'lodash';
import ContactCard from '../ContactCard';
import { contact } from '../../../config/jest/mock-data';

describe('Testing ContactCard component', () => {
  it('renders ContactCard using Snapshots', () => {
    expect(renderer.create(
      <MuiThemeProvider>
        <ContactCard
          contact={contact}
          onEditClick={jest.fn}
          onDeleteClick={jest.fn}
        />
      </MuiThemeProvider>
    )).toMatchSnapshot();
  });
  it('renders ContactCard omit phone using Snapshots', () => {
    expect(renderer.create(
      <MuiThemeProvider>
        <ContactCard
          contact={_.omit(contact, 'phoneNumber')}
          onEditClick={jest.fn}
          onDeleteClick={jest.fn}
        />
      </MuiThemeProvider>
    )).toMatchSnapshot();
  });
  it('renders ContactCard omit imgUrl using Snapshots', () => {
    expect(renderer.create(
      <MuiThemeProvider>
        <ContactCard
          contact={_.omit(contact, 'imgUrl')}
          onEditClick={jest.fn}
          onDeleteClick={jest.fn}
        />
      </MuiThemeProvider>
    )).toMatchSnapshot();
  });
});
