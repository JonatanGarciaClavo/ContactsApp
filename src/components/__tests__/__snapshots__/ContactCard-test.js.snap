exports[`Testing ContactCard component renders ContactCard omit imgUrl using Snapshots 1`] = `
<div
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": "2px",
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
             0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "fontFamily": "Roboto, sans-serif",
      "margin": "1em 0 0 0",
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "width": "300px",
      "zIndex": 1,
    }
  }>
  <div
    style={
      Object {
        "paddingBottom": 0,
      }
    }>
    <div
      style={
        Object {
          "muiPrepared": true,
          "position": "relative",
        }
      }>
      <div
        style={
          Object {
            "muiPrepared": true,
          }
        }>
        <img
          role="presentation"
          src="http://i.imgur.com/mbZIBzc.png"
          style={
            Object {
              "maxWidth": "100%",
              "minWidth": "100%",
              "muiPrepared": true,
              "verticalAlign": "top",
              "width": "100%",
            }
          } />
      </div>
      <div
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "muiPrepared": true,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }>
        <div
          style={
            Object {
              "height": "100%",
              "muiPrepared": true,
              "position": "relative",
            }
          }>
          <div
            style={
              Object {
                "background": "rgba(0, 0, 0, 0.54)",
                "bottom": 0,
                "left": 0,
                "muiPrepared": true,
                "paddingTop": 8,
                "position": "absolute",
                "right": 0,
              }
            }>
            <div
              style={
                Object {
                  "muiPrepared": true,
                  "padding": 16,
                  "position": "relative",
                }
              }>
              <span
                style={
                  Object {
                    "color": "rgba(255, 255, 255, 0.87)",
                    "display": "block",
                    "fontSize": 24,
                    "lineHeight": "36px",
                    "muiPrepared": true,
                  }
                }>
                Tester
              </span>
              <span
                style={
                  Object {
                    "color": "rgba(255, 255, 255, 0.54)",
                    "display": "block",
                    "fontSize": 14,
                    "muiPrepared": true,
                  }
                } />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "muiPrepared": true,
          "padding": 16,
          "position": "relative",
        }
      }>
      <span
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "block",
            "fontSize": "12pt",
            "lineHeight": "36px",
            "muiPrepared": true,
          }
        }>
        test@test.com
      </span>
      <span
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "display": "block",
            "fontSize": 14,
            "muiPrepared": true,
          }
        }>
        666666666
      </span>
    </div>
    <div
      style={
        Object {
          "muiPrepared": true,
          "padding": 8,
          "position": "relative",
        }
      }>
      <button
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": 0,
            "marginRight": 8,
            "minWidth": 88,
            "muiPrepared": true,
            "outline": "none",
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transform": "translate(0, 0)",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "verticalAlign": null,
          }
        }
        tabIndex={0}
        type="button">
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}>
          <span
            style={
              Object {
                "fontSize": 14,
                "fontWeight": 500,
                "letterSpacing": 0,
                "muiPrepared": true,
                "paddingLeft": 16,
                "paddingRight": 16,
                "position": "relative",
                "textTransform": "uppercase",
                "verticalAlign": "middle",
              }
            }>
            Edit
          </span>
        </div>
      </button>
      <button
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": 0,
            "marginRight": 8,
            "minWidth": 88,
            "muiPrepared": true,
            "outline": "none",
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transform": "translate(0, 0)",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "verticalAlign": null,
          }
        }
        tabIndex={0}
        type="button">
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}>
          <span
            style={
              Object {
                "fontSize": 14,
                "fontWeight": 500,
                "letterSpacing": 0,
                "muiPrepared": true,
                "paddingLeft": 16,
                "paddingRight": 16,
                "position": "relative",
                "textTransform": "uppercase",
                "verticalAlign": "middle",
              }
            }>
            Delete
          </span>
        </div>
      </button>
    </div>
  </div>
</div>
`;

exports[`Testing ContactCard component renders ContactCard omit phone using Snapshots 1`] = `
<div
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": "2px",
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
             0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "fontFamily": "Roboto, sans-serif",
      "margin": "1em 0 0 0",
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "width": "300px",
      "zIndex": 1,
    }
  }>
  <div
    style={
      Object {
        "paddingBottom": 0,
      }
    }>
    <div
      style={
        Object {
          "muiPrepared": true,
          "position": "relative",
        }
      }>
      <div
        style={
          Object {
            "muiPrepared": true,
          }
        }>
        <img
          role="presentation"
          src="http://www.test.image.com"
          style={
            Object {
              "maxWidth": "100%",
              "minWidth": "100%",
              "muiPrepared": true,
              "verticalAlign": "top",
              "width": "100%",
            }
          } />
      </div>
      <div
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "muiPrepared": true,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }>
        <div
          style={
            Object {
              "height": "100%",
              "muiPrepared": true,
              "position": "relative",
            }
          }>
          <div
            style={
              Object {
                "background": "rgba(0, 0, 0, 0.54)",
                "bottom": 0,
                "left": 0,
                "muiPrepared": true,
                "paddingTop": 8,
                "position": "absolute",
                "right": 0,
              }
            }>
            <div
              style={
                Object {
                  "muiPrepared": true,
                  "padding": 16,
                  "position": "relative",
                }
              }>
              <span
                style={
                  Object {
                    "color": "rgba(255, 255, 255, 0.87)",
                    "display": "block",
                    "fontSize": 24,
                    "lineHeight": "36px",
                    "muiPrepared": true,
                  }
                }>
                Tester
              </span>
              <span
                style={
                  Object {
                    "color": "rgba(255, 255, 255, 0.54)",
                    "display": "block",
                    "fontSize": 14,
                    "muiPrepared": true,
                  }
                } />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "muiPrepared": true,
          "padding": 16,
          "position": "relative",
        }
      }>
      <span
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "block",
            "fontSize": "12pt",
            "lineHeight": "36px",
            "muiPrepared": true,
          }
        }>
        test@test.com
      </span>
      <span
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "display": "block",
            "fontSize": 14,
            "muiPrepared": true,
          }
        }>
        Without phone
      </span>
    </div>
    <div
      style={
        Object {
          "muiPrepared": true,
          "padding": 8,
          "position": "relative",
        }
      }>
      <button
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": 0,
            "marginRight": 8,
            "minWidth": 88,
            "muiPrepared": true,
            "outline": "none",
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transform": "translate(0, 0)",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "verticalAlign": null,
          }
        }
        tabIndex={0}
        type="button">
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}>
          <span
            style={
              Object {
                "fontSize": 14,
                "fontWeight": 500,
                "letterSpacing": 0,
                "muiPrepared": true,
                "paddingLeft": 16,
                "paddingRight": 16,
                "position": "relative",
                "textTransform": "uppercase",
                "verticalAlign": "middle",
              }
            }>
            Edit
          </span>
        </div>
      </button>
      <button
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": 0,
            "marginRight": 8,
            "minWidth": 88,
            "muiPrepared": true,
            "outline": "none",
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transform": "translate(0, 0)",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "verticalAlign": null,
          }
        }
        tabIndex={0}
        type="button">
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}>
          <span
            style={
              Object {
                "fontSize": 14,
                "fontWeight": 500,
                "letterSpacing": 0,
                "muiPrepared": true,
                "paddingLeft": 16,
                "paddingRight": 16,
                "position": "relative",
                "textTransform": "uppercase",
                "verticalAlign": "middle",
              }
            }>
            Delete
          </span>
        </div>
      </button>
    </div>
  </div>
</div>
`;

exports[`Testing ContactCard component renders ContactCard using Snapshots 1`] = `
<div
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": "2px",
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
             0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "fontFamily": "Roboto, sans-serif",
      "margin": "1em 0 0 0",
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "width": "300px",
      "zIndex": 1,
    }
  }>
  <div
    style={
      Object {
        "paddingBottom": 0,
      }
    }>
    <div
      style={
        Object {
          "muiPrepared": true,
          "position": "relative",
        }
      }>
      <div
        style={
          Object {
            "muiPrepared": true,
          }
        }>
        <img
          role="presentation"
          src="http://www.test.image.com"
          style={
            Object {
              "maxWidth": "100%",
              "minWidth": "100%",
              "muiPrepared": true,
              "verticalAlign": "top",
              "width": "100%",
            }
          } />
      </div>
      <div
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "muiPrepared": true,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }>
        <div
          style={
            Object {
              "height": "100%",
              "muiPrepared": true,
              "position": "relative",
            }
          }>
          <div
            style={
              Object {
                "background": "rgba(0, 0, 0, 0.54)",
                "bottom": 0,
                "left": 0,
                "muiPrepared": true,
                "paddingTop": 8,
                "position": "absolute",
                "right": 0,
              }
            }>
            <div
              style={
                Object {
                  "muiPrepared": true,
                  "padding": 16,
                  "position": "relative",
                }
              }>
              <span
                style={
                  Object {
                    "color": "rgba(255, 255, 255, 0.87)",
                    "display": "block",
                    "fontSize": 24,
                    "lineHeight": "36px",
                    "muiPrepared": true,
                  }
                }>
                Tester
              </span>
              <span
                style={
                  Object {
                    "color": "rgba(255, 255, 255, 0.54)",
                    "display": "block",
                    "fontSize": 14,
                    "muiPrepared": true,
                  }
                } />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "muiPrepared": true,
          "padding": 16,
          "position": "relative",
        }
      }>
      <span
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "block",
            "fontSize": "12pt",
            "lineHeight": "36px",
            "muiPrepared": true,
          }
        }>
        test@test.com
      </span>
      <span
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "display": "block",
            "fontSize": 14,
            "muiPrepared": true,
          }
        }>
        666666666
      </span>
    </div>
    <div
      style={
        Object {
          "muiPrepared": true,
          "padding": 8,
          "position": "relative",
        }
      }>
      <button
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": 0,
            "marginRight": 8,
            "minWidth": 88,
            "muiPrepared": true,
            "outline": "none",
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transform": "translate(0, 0)",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "verticalAlign": null,
          }
        }
        tabIndex={0}
        type="button">
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}>
          <span
            style={
              Object {
                "fontSize": 14,
                "fontWeight": 500,
                "letterSpacing": 0,
                "muiPrepared": true,
                "paddingLeft": 16,
                "paddingRight": 16,
                "position": "relative",
                "textTransform": "uppercase",
                "verticalAlign": "middle",
              }
            }>
            Edit
          </span>
        </div>
      </button>
      <button
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": 0,
            "marginRight": 8,
            "minWidth": 88,
            "muiPrepared": true,
            "outline": "none",
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transform": "translate(0, 0)",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "verticalAlign": null,
          }
        }
        tabIndex={0}
        type="button">
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}>
          <span
            style={
              Object {
                "fontSize": 14,
                "fontWeight": 500,
                "letterSpacing": 0,
                "muiPrepared": true,
                "paddingLeft": 16,
                "paddingRight": 16,
                "position": "relative",
                "textTransform": "uppercase",
                "verticalAlign": "middle",
              }
            }>
            Delete
          </span>
        </div>
      </button>
    </div>
  </div>
</div>
`;
